csv/generic/shared/src/test/scala/fs2/data/csv/generic/CsvRowDecoderTest.scala

```diff
@@ -82,4 +87,161 @@ object CsvRowDecoderTest extends SimpleIOSuite {
       expect(testOptionalStringDecoder(csvRowEmptyCell) == Right(TestOptionalString(1, None, 42)))
   }
 
+  implicit val decoderErrorEq: Eq[Throwable] = (a, b) => a.toString === b.toString
+
+  implicit val decoderTestDataEq: Eq[TestData] = (a, b) => a == b
+
+  test("Parser should return all decoder results as values when using attemptDecodeUsingHeaders") {
+    val content =
+      """name,age,description
+        |John Doe,47,description 1
+        |Jane Doe,50
+        |Bob Smith,80,description 2
+        |Alice Grey,78
+        |""".stripMargin
+
+    val expected: List[Either[Throwable, TestData]] = List(
+      Right(TestData("John Doe", "47", "description 1")),
+      Left(new CsvException("Headers have size 3 but row has size 2. Both numbers must match! in line 3", None)),
+      Right(TestData("Bob Smith", "80", "description 2")),
+      Left(new CsvException("Headers have size 3 but row has size 2. Both numbers must match! in line 5", None))
+    )
+
+    Stream
+      .emit(content)
+      .covary[IO]
```

Comment:
You don't need `IO` in these tests, `Fallible` will do just fine as we don't have to read any files.